File: java/rest-simple/src/main/java/com/example/restsimple/response/StudentListResponse.java
package com.example.restsimple.response;

import com.example.restsimple.model.Student;

import jakarta.validation.constraints.NotNull;
import java.util.List;

public class StudentListResponse {
    @NotNull(message = "students is required")
    public List<Student> students;

    public StudentListResponse(List<Student> students) {
        this.students = students;
    }
}

File: kotlin/rest-simple/src/main/kotlin/com/example/restsimple/response/StudentListResponse.kt
package com.example.restsimple.response

import com.example.restsimple.model.Student
import javax.validation.constraints.NotNull

class StudentListResponse(students: List<Student?>?) {
    var students: @NotNull(message = "students is required") MutableList<Student?>? = students?.toMutableList()
}
